tesseract_kinematics/test/kinematics_core_unit.cpp

```diff
@@ -89,6 +91,25 @@ TEST(TesseractKinematicsUnit, UtilsHarmonizeUnit)  // NOLINT
   EXPECT_NEAR(q[1], 3 * M_PI_4, 1e-6);
 }
 
+TEST(TesseractKinematicsUnit, UtilsNearSingularityUnit)  // NOLINT
+{
+  tesseract_scene_graph::SceneGraph::Ptr scene_graph = tesseract_kinematics::test_suite::getSceneGraphABB();
+
+  tesseract_kinematics::KDLFwdKinChain fwd_kin;
+  fwd_kin.init(scene_graph, "base_link", "tool0", "manip");
+
+  Eigen::VectorXd jv = Eigen::VectorXd::Zero(6);
+  Eigen::MatrixXd jacobian = fwd_kin.calcJacobian(jv);
+  EXPECT_TRUE(tesseract_kinematics::isNearSingularity(jacobian, 0.001));
+  jv[4] = 1 * M_PI / 180.0;
+  jacobian = fwd_kin.calcJacobian(jv);
+  EXPECT_TRUE(tesseract_kinematics::isNearSingularity(jacobian));
+  jv[4] = 2 * M_PI / 180.0;
+  jacobian = fwd_kin.calcJacobian(jv);
+  EXPECT_FALSE(tesseract_kinematics::isNearSingularity(jacobian));
```

Comment:
What are you trying to show by rotating joint 5 here? I'm a little confused about what you're testing here

Reply:
For the robot being used, singularity occurs when joint 4,5,6 are all at zeros. Changing joint 5 value should move it out of singularity which is demonstrated by the unit test. As joint 5 increase so does the smallest singular value.

Reply:
That's what I don't understand. Axes 4 and 6 would still be aligned (i.e. still in singularity) when J5=0, J5=180, and J5=360. This unit test shows that there is no singularity when J5=360, which should be the same condition as J5=0.

Reply:
Sorry, I don't see where J5 is being set to 360 deg. It currently is only setting it to 0, 1 deg and 2 deg.

Reply:
My bad; I was reading this wrong. This makes sense now